pkg/util/hlc/hlc.go

```diff
@@ -217,39 +222,50 @@ func (c *Clock) MaxOffset() time.Duration {
 
 // getPhysicalClockAndCheck locks mu in order to access the physical clock, check for
 // time jumps and update the internal jump checking state.
-func (c *Clock) getPhysicalClockAndCheck() int64 {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-	return c.getPhysicalClockAndCheckLocked()
+func (c *Clock) getPhysicalClockAndCheck(ctx context.Context) int64 {
+	oldTime := atomic.LoadInt64(&c.lastPhysicalTime)
+	newTime := c.physicalClock()
+	lastPhysTime := oldTime
+	for {
+		atomic.CompareAndSwapInt64(&c.lastPhysicalTime, lastPhysTime, newTime)
```

Comment:
Micro-optimization: This implementation always does at least two loads and one CAS. An easy improvement is to break out of the loop immediately if CAS returns true; this saves one atomic load in the fastest case.